Source/ACE/Entity/WorldObject.cs

```diff
@@ -2464,16 +2464,21 @@ public void SerializePhysicsData(BinaryWriter writer)
                     }
                     else
                     {
+                        // Adding these debug lines - don't think we can hit these, but want to make sure. Og II
+                        log.Debug($"Our flag is set but we have no data length. {this.Guid.Full:X}");
                         writer.Write(0u);
                     }
                 }
                 else
-                {                  
+                {
+                    log.Debug($"Our flag is set but our current motion state is null. {this.Guid.Full:X}");
                     writer.Write(0u);
                 }
             }
             else if ((PhysicsDescriptionFlag & PhysicsDescriptionFlag.AnimationFrame) != 0)
                 writer.Write((AnimationFrame ?? 0u));
+            // TODO: Keep an eye on this, are we sure the client does not just ignore it?   I would think they way it reads by buffer length that this would blow up.
```

Comment:
the else if is how the client parses the code.  Its funky.  The above movement data writing code is wrong.  I'll try to put a patch in for it this week.